### DISCUSSION TITLE
moveout correlation

### DISCUSSION THREAD

**Discussion by JouSF-1409** [xumi1993/seispy#54] [Q&A]
您好，我是刚刚开始做接收函数这一领域的学生。刚读完代码的一部分。
想知道里面动校正的算法怎么实现，能不能给出数学证明，我准备做的深度比较大，在想能不能直接化用，需不需要考虑地球曲率的问题。
我也很想知道软件的设计结构，git里的注释和介绍文档不是很清楚，我该从哪个文档开始读？

**Comment by xumi1993** [comment]
目前这个软件包里的后处理算法都是基于水平层状假设的，包括moveout correction，这部分原理可以参考 introduction to seismology的相应章节。

如果你觉得从文档中依然难以获得想要的信息，那么十分抱歉，我个人精力有限，导致文档难以维护。文档中的教程调用了程序的模块，如果你想学习的话可以查看相应代码。

**Reply by xumi1993** [reply]
补充一点，我们在求解Ps到时时考虑了展平变换，无法计算深度较大的震相，但可以计算地幔转换带震相

**Reply by JouSF-1409** [reply]
想用seispy做S波接收函数，主要是考虑到穿刺点距离较远，结合一些比较差的区域成像结果，在考虑是否会引入可观的误差
前人的代码是不考虑这些问题的，希望不会有太大的影响。

**Reply by xumi1993** [reply]
seispy 对 S波接收函数缺少测试，可以保证正确计算出接收函数。
